doc/rtd/reference/datasources/lxd.rst

```diff
@@ -37,26 +36,16 @@ launched container, use either LXD profiles or
 ``lxc launch ... -c <key>="<value>"`` at initial container launch, by setting
 one of the following keys:
 
-- ``user.meta-data``: YAML metadata which will be appended to base metadata.
-- ``user.vendor-data``: YAML which overrides any metadata values.
-- ``user.network-config``: YAML representing either :ref:`network_config_v1` or
+- ``cloud-init.vendor-data``: YAML which overrides any metadata values.
+- ``cloud-init.network-config``: YAML representing either :ref:`network_config_v1` or
   :ref:`network_config_v2` format.
-- ``user.user-data``: YAML which takes precedence and overrides both metadata
+- ``cloud-init.user-data``: YAML which takes precedence and overrides both metadata
   and vendor data values.
 - ``user.any-key``: Custom user configuration key and value pairs, which can be
```

Comment:
I was unable to find anything that corroborates this claim. I couldn't find any mention of `user.any-kay` it in the code, nor do I see the lxd datasource storing arbitrary keys anywhere that could pass through.

I tested the `cloud-init query` portion of this by setting `user.any-key` to `key: value` and cleaning the instance then rebooting. I didn't test the instance-data template portion, but I don't have high hopes.

Think we should remove it @TheRealFalcon @blackboxsw?

Reply:
> ```
> $ ': lxc launch ubuntu-daily:jammy -c user.my-tag=neato test-custom-user-config
> $ lxc exec test-custom-user-config -- cloud-init query ds.config.user_my_tag
> neato
> # jinja template custom behavior
> $ lxc exec test-custom-user-config -- cloud-init query --format "{% if ds.config.user_my_tag == 'neato'  %}sh /custom_script.sh{% endif %}"
> /custom_script.sh
> ```
> 
> Key aliases are provided by cloud-init for ease of use in jinja templates and queries
> 
> ```
> $ cloud-init query --format "{{ds.config}}"
> {'user.my-key': 'two', 'user_my_key': 'two'}
> ```

Thanks for the clarification. Reading this doc, I interpreted it as saying that key `user.any-key` is literal, not arbitrary, and expected key pairs set in `user.any-key` to be the key/values that would be exposed / visible in `cloud-init query` and in template/jinja. I see now that `user.any-key` _is_ a key. I'll modify it to hopefully avoid others from interpreting it like I did.